aws-crt/CRT.cs

```diff
@@ -297,42 +340,48 @@ public override LibraryHandle LoadLibrary(string name)
                     Assembly crtAsm = Assembly.GetAssembly(typeof(CRT));
                     path = crtAsm.Location.Replace(crtAsm.GetName().Name + ".dll", name);
                 }
-                return new LibraryHandle(dl.dlopen(path, dl.RTLD_NOW), path);
+                return new LibraryHandle(darwin.dlopen(path, darwin.RTLD_NOW), path);
             }
 
             public override void FreeLibrary(IntPtr handle)
             {
-                dl.dlclose(handle);
+                darwin.dlclose(handle);
             }
-            
+
             public override IntPtr GetFunction(IntPtr handle, string name)
             {
-                return dl.dlsym(handle, name);
+                return darwin.dlsym(handle, name);
             }
 
             public override string GetLastError()
             {
-                return dl.dlerror();
+                return darwin.dlerror();
             }
         }
 
         private static PlatformLoader s_loader = null;
-        internal static PlatformLoader Loader 
+        internal static PlatformLoader Loader
         {
             get {
                 if (s_loader != null)
                 {
                     return s_loader;
                 }
 
-                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
This isn't available for `net35/net45` - we do have a shim in `Aws.Crt.Platform.GetRuntimePlatformOS()`.

(looks like Windows CI is running `dotnet build -f netstandard2.0` - unsure if the images are set up for it, but exercising .NET Framework here too might be nice)

Reply:
Oh, I can detect windows another way, good call.